### PR TITLE
Add DNS cleaner to Terraform

### PR DESCRIPTION
**Pull request by Sawthis**
**Description**

Changes proposed in this pull request:

- Add DNS cleaner to Terraform
- Remove not used dnscleaner app

**Related issue(s)**

https://github.com/kyma-project/test-infra/issues/7609

### CODE REVIEW

File: configs/terraform/environments/prod/dns-cleaner.tf [hunk 1]
```diff
@@ -0,0 +1,22 @@
+resource "google_service_account" "sa_gke_kyma_integration" {
+  account_id   = "sa_gke_kyma_integration@sap-kyma-prow.iam.gserviceaccount.com"
+  display_name = "sa_gke_kyma_integration"
```
Comment (dekiel) [inline configs/terraform/environments/prod/dns-cleaner.tf:3]:
```suggestion
  account_id   = "sa-gke-kyma-integration@sap-kyma-prow.iam.gserviceaccount.com"
  display_name = "sa-gke-kyma-integration@sap-kyma-prow.iam.gserviceaccount.com"
```

You can't use underscore for sa names. See terraform plan error.

```
Error: "account_id" ("sa_gke_kyma_integration@sap-kyma-prow.iam.gserviceaccount.com") doesn't match regexp "^[a-z](?:[-a-z0-9]{4,28}[a-z0-9])$"
```

File: configs/terraform/environments/prod/dns-cleaner.tf [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+}
+
+resource "google_project_iam_binding" "view_container_analysis_ccurrences" {
```
Comment (dekiel) [inline configs/terraform/environments/prod/dns-cleaner.tf:6]: The resource name sounds to generic. If another identity will need to get same IAM role assigned we will have to use different resource name than `view_container_analysis_ccurrences`. The name of resource should be more specific and provide more context. Maybe something like.

```suggestion
resource "google_project_iam_binding" "dns_cleaner_view_container_analysis_occurrences" {
```

This is the same for bindings of `dns_reader` and `bucket_get` roles.

**Review by dekiel — CHANGES_REQUESTED** [review]
Please add comments for which application this config exist and which workload is using it. A link to the cleaner code etc.

**Review by dekiel — APPROVED** [review]

**Review by IwonaLanger — APPROVED** [review]

**Pull request merged**